### DISCUSSION TITLE
UI support for Apache Polaris

### DISCUSSION THREAD

**Discussion by MonkeyCanCode** [apache/polaris#306] [Ideas]
As of today, both [Nessie](https://projectnessie.org/guides/ui/) and [Unity Catalog](https://github.com/unitycatalog/unitycatalog/tree/main/ui) offer user interfaces (UIs) for both open source and enterprise editions. In contrast, Apache Polaris currently only supports a UI through the [Snowflake service](https://www.snowflake.com/en/blog/polaris-catalog-open-source/), with no UI available for the open source version. Are there any plans to address this gap?

Regarding Apache Amoro, they are awaiting the initial release of Apache Polaris before considering integration support, particularly if we do not develop our own UI (see: apache/amoro#3089).

If we decide to develop our own UI to match the feels and looks of the enterprise edition, are there any preferred programming languages or frameworks that we should use before submitting a pull request?

**Comment by lamb-russell** [comment]
i am in the process of writing my own Electron GUI that uses the CLI as the backend.   Please feel free to use and/or contribute. https://github.com/lamb-russell/polaris-electron-app

**Comment by RussellSpitzer** [comment]
I'm open to including a UI in the project, so any proposals would be welcome

**Reply by lamb-russell** [reply]
I can add a few more features to the UI I'm building. It currently has implemented the basic features of the CLi: eg, list, create, delete, grant, revoke. So far catalog, principal, and principal role are implemented. if you think it could be a good UI for Polaris i can either link to the repo in the docs or even add it to the code base.

**Comment by jbonofre** [comment]
I think it makes sense to have an UI in the project.
I would go with React + Ant design as first layer, pretty aligned with what's coming in Polaris.
I can draft something around that (I did several admin consoles using this).

A first step would be to manage the entities (tables and views) via the UI (creating/updating/removing entities), the RBAC model (managing principals and roles) and also preparing additional management layer (like the maintenance notification layer we are discussing).

**Comment by lamb-russell** [comment]
JB good idea to use Ant for design. So far I got managing several entities
(catalog, principal, principal role, catalog role) into the react gui.

**Comment by akhilputhiry** [comment]
There is a project called [nimtable](https://github.com/nimtable/nimtable)

**Reply by michal-kuzak** [reply]
> There is a project called [nimtable](https://github.com/nimtable/nimtable)

you can view polaris catalogs in nimtable, but you can not administer them, create new catalogs or manage privileges, principals, auth, keys etc

**Comment by hongbo-miao** [comment]
Besides [Nimtable](https://github.com/nimtable/nimtable), there is another one [Polaris Console](https://github.com/apache/polaris-tools/blob/main/console/README.md), but Polaris Console is still early, missing both Docker image and Helm chart.

**Comment by jbonofre** [comment]
It's not so early, but it needs a first release.

I proposed to move forward on the first Polaris Console release on the mailing list: https://lists.apache.org/thread/trtvk39hx91o7g4wdbbhcq5o5mhypx0d

I would like to propose to move forward on Polaris Console release, including documentation and communicate more about that.

By the way, we love contribution, so feel free to create issues and PR for the Polaris Console.
